app/src/main/java/com/duckduckgo/app/history/SaveToHistory.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.history
+
+import com.duckduckgo.app.browser.DuckDuckGoUrlDetector
+import com.duckduckgo.app.di.AppCoroutineScope
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+interface SaveToHistory {
```

Comment:
I see we have this interface here that is way simpler that the other one, and this is exactly what I was thinkings about ⭐ 

So question is why 2 interfaces, and which one is the one feature like autocomplete will use.

Reply:
The way I see them, only this one should belong to an api module (if we decide to create a module for this feature) and therefore be the interface autocomplete and other features will use. As to why having 2 interfaces, it's just a way to separate business logic (usecase/interactor) and data access (repository), in the same way AutoComplete class handles all the autocomplete logic and deals with SavedSitesRepository (and HistoryRepository as a a next step) for data access

Reply:
ok then let's propose a new module for history and open a proposal for api. That way  1) we will have a clear separation of what's public/internal, 2) we will have a more focused discussion on what to expose in the interface

Reply:
As discussed on MM, a new module has been added. See https://github.com/duckduckgo/Android/pull/4416